src/Symfony/Component/Security/Core/Authentication/RememberMe/CacheTokenVerifier.php

```diff
@@ -60,9 +62,18 @@ public function updateExistingToken(PersistentTokenInterface $token, string $tok
     {
         // When a token gets updated, persist the outdated token for $outdatedTokenTtl seconds so we can
         // still accept it as valid in verifyToken
-        $item = $this->cache->getItem($this->cacheKeyPrefix.$token->getSeries());
+        $item = $this->cache->getItem($this->getCacheKey($token));
         $item->set($token->getTokenValue());
         $item->expiresAfter($this->outdatedTokenTtl);
         $this->cache->save($item);
     }
+
+    private function getCacheKey(PersistentTokenInterface $token): string
+    {
+        return $this->cacheKeyPrefix.preg_replace(
```

Comment:
We usually use `rawurlencode()` to encode cache keys, that's a super fast function, especially compared to PCRE.
It will also make things unambiguous, while a conditional search and replace can create collisions.
If you agree with using `rawurlencode()`, then I'd suggest removing the private method.

Reply:
done